src/properties.js

```diff
@@ -458,6 +458,55 @@ export function suspendableProp(options) {
     )
 }
 
+export function delayableHandler(options) {
+  /*
+  Delays `[handler]` calls until after `[filter]` become true.
+  */
+  const handler = options.handler
+  const filter = options.filter
+  return (Component) =>
+    setWrapperName(
+      Component,
+      class delayableHandler extends BaseComponent {
+        constructor(props) {
+          super(props)
+          this.state = {
+            shouldTrigger: false,
+            value: props.value,
+          }
+          this.trigger = () => {
+            const {
+              props: { [handler]: triggerFunction, [filter]: value },
+            } = this
+            if (value) {
+              triggerFunction()
+              this.setState({ shouldTrigger: false })
```

Comment:
Should check if `shouldTrigger` is already `false` first to minimize unnecessary updates.

Reply:
Also checked below if `shouldTrigger` is not already `true`